cxx-squid/src/test/java/org/sonar/cxx/CxxConfigurationTest.java

```diff
@@ -177,7 +177,7 @@ public void shouldHandleSpeciificV120OptionsCorrectly() {
     
     assertThat(config.getIncludeDirectories().size()).isEqualTo(0);    
     List<String> defines = config.getDefines();
-    assertThat(defines.size()).isEqualTo(commonDefines + 8);
```

Comment:
I moved some defines to ParseCommonCompilerOptions() and the compiler version specific part is defined individually e.g.:
```
    AddMacro("__cplusplus=199711L", fileElement);
    if ... AddMacro("__cplusplus_winrt=201009", fileElement);
    AddMacro("_MSC_VER=1900", fileElement);
    AddMacro("_MSC_FULL_VER=190022816", fileElement);
    AddMacro("_MFC_VER=0x0C00", fileElement);   
```
The parameter parsing must not be separated as long there is no conflict.
The define for ```"_M_ARM_FP"``` is removed (see other comment).